aws-datastore/src/main/java/com/amplifyframework/datastore/DataStoreConfiguration.java

```diff
@@ -0,0 +1,340 @@
+                        break;
+                    case SYNC_PAGE_SIZE:
+                        builder.syncPageSize(pluginJson.getInt(ConfigKey.SYNC_PAGE_SIZE.toString()));
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Unsupported config key = " + configKey.toString());
+                }
+            } catch (JSONException jsonException) {
+                throw new DataStoreException(
+                        "Issue encountered while parsing configuration JSON",
+                        jsonException, "Ensure your amplifyconfiguration.json is valid."
+                );
+            }
+        }
+
+        return builder;
+    }
+
+    /**
+     * Creates an {@link DataStoreConfiguration} which uses all default values.
+     * @return A default {@link DataStoreConfiguration}
+     */
+    @NonNull
+    public static DataStoreConfiguration defaults() {
+        return builder().build();
+    }
+
+    /**
+     * Gets the data store error handler.
+     * @return Data store error handler.
+     */
+    @NonNull
+    public DataStoreErrorHandler getDataStoreErrorHandler() {
+        return this.dataStoreErrorHandler;
+    }
+
+    /**
+     * Gets the data store conflict handler.
+     * @return Data store conflict handler
+     */
+    @NonNull
+    public DataStoreConflictHandler getDataStoreConflictHandler() {
+        return this.dataStoreConflictHandler;
+    }
+
+    /**
+     * Get the sync interval. The sync interval is the amount of time after a base sync, during which
+     * the optimized delta-sync may be requested, instead of a full base sync.
+     * @return The sync interval
+     */
+    @IntRange(from = 0)
+    public long getSyncIntervalInMinutes() {
+        return this.syncIntervalInMinutes;
+    }
+
+    /**
+     * Gets the maximum number of records that the client wants to process, while it is requesting
+     * a base/delta sync operation from AppSync.
+     * @return The max number of records to process from AppSync.
+     */
+    @IntRange(from = 0)
+    public int getSyncMaxRecords() {
+        return this.syncMaxRecords;
+    }
+
+    /**
+     * Gets the number of items that should be requested in page, from AppSync, during
+     * a sync operation.
+     * @return Desired size of a page of results from an AppSync sync response
+     */
+    @IntRange(from = 0)
+    public int getSyncPageSize() {
+        return this.syncPageSize;
+    }
+
+    /**
+     * Builds instances of {@link AWSDataStorePlugin} by providing a variety of
+     * configuration methods.
+     */
+    public static final class Builder {
+        private DataStoreErrorHandler dataStoreErrorHandler;
+        private DataStoreConflictHandler dataStoreConflictHandler;
+        private Long syncIntervalMs;
+        private Integer syncMaxRecords;
+        private Integer syncPageSize;
```

Comment:
I don't think you ever use the fact that these are objects. (They have non-null default values, and you overwrite the m via primitive types.)

So, I think it should be both safe and preferable to store these as their primitive counterparsts,`long` and `int` .